include/dali/core/byte_io.h

```diff
@@ -34,6 +37,19 @@ void ReadValueImpl(T &value, const uint8_t* data) {
   value >>= pad;
 }
 
+template <int nbytes,
+          bool is_little_endian,
+          typename T,
+          std::enable_if_t<std::is_enum<T>::value, bool> = true>
+void ReadValueImpl(T &value, const uint8_t* data) {
+  using U = std::underlying_type_t<T>;
+  static_assert(nbytes == sizeof(U),
+    "nbytes is expected to be the same as sizeof(std::underlying_type_t<T>)");
```

Comment:
This does not need to be the case. The underlying type may be oversized for the enum and we might be deliberately reading a smaller representation.
```suggestion
  static_assert(nbytes <= sizeof(U),
    "`nbytes` should not exceed the size of the underlying type of the enum");
```